index.js

```diff
@@ -12,10 +14,41 @@ function promisifyOwnProperties(obj) {
 
       return { [entry[0]]: entry[1] };
     })
-    .reduce((acc, curr) => Object.assign(acc, curr), {});
+    .reduce(combiner, {});
 }
 
 module.exports = obj => {
+  if (!obj) {
+    return obj;
+  }
+
+  if (typeof obj !== 'function' && typeof obj !== 'object') {
+    return obj;
+  }
+
+  if (Array.isArray(obj)) {
+    const props = promisifyOwnProperties(entries(obj).filter(e => obj.indexOf(e[1]) === -1).reduce(combiner, {}));
```

Comment:
i wonder if this would be easier as `const props = entries(obj).filter(e => obj.indexOf(e[1]) === -1).map(x => promisify(x));` - then you'd do `props.length` below, and `props.forEach`, without the need to make an object?